### PR TITLE
Add logs to PullRequest Actor

### PR DESCRIPTION
**Pull request by AlitzelMendez**
In order to gather more information for the issue: Some dependency updates are intermittently failing due to timeouts in service fabric actors (https://github.com/dotnet/arcade/issues/7262) adding more logs

### CODE REVIEW

File: src/Maestro/SubscriptionActorService/PullRequestActor.cs [hunk 1]
```diff
@@ -1207,6 +1224,7 @@ private async Task UpdatePullRequestAsync(InProgressPullRequest pr, List<UpdateA
             string targetRepository,
             string branch)
         {
+            Logger.LogInformation($"Get Required Updates from {branch} to {targetRepository}");
```
Comment (riarenas) [inline src/Maestro/SubscriptionActorService/PullRequestActor.cs:1227]: [nit]
```suggestion
            Logger.LogInformation($"Getting Required Updates from {branch} to {targetRepository}");
```

Most other log statements are written in this tense, as they indicate that an operation is ongoing

File: src/Maestro/SubscriptionActorService/PullRequestActor.cs [hunk 2]
```diff
@@ -1288,6 +1307,7 @@ private async Task UpdatePullRequestAsync(InProgressPullRequest pr, List<UpdateA
                 requiredUpdates.Add((coherencyUpdateParameters, coherencyUpdates.ToList()));
             }
 
+            Logger.LogInformation("Finished get Required Updates from {branch} to {targetRepository}");
```
Comment (riarenas) [inline src/Maestro/SubscriptionActorService/PullRequestActor.cs:1310]: [nit]
```suggestion
            Logger.LogInformation("Finished getting Required Updates from {branch} to {targetRepository}");
```

**Review by riarenas — APPROVED** [review]
LGTM some minor wording nits and requests for additional info to be added to the statements when it makes sense.

File: src/Maestro/SubscriptionActorService/PullRequestActor.cs [hunk 3]
```diff
@@ -386,10 +391,12 @@ public virtual async Task<(InProgressPullRequest pr, bool canUpdate)> Synchroniz
         [ActionMethod("Synchronizing Pull Request: '{url}'")]
         private async Task<ActionResult<SynchronizePullRequestResult>> SynchronizePullRequestAsync(string prUrl)
         {
+            Logger.LogInformation($"Synchronizing Pull Request {prUrl}");
```
[the remark this answers is not in the document]
Comment (AlitzelMendez) [reply src/Maestro/SubscriptionActorService/PullRequestActor.cs:394]: do you know where this is logged? I was not able to find it on AI

**Pull request merged**